Tests/src/PublicAPITests/UserSystemTests.cpp

```diff
@@ -1020,18 +1024,22 @@ CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, AgeNotVerifiedTest)
 	csp::common::String UserId;
 
 	// False Log in
-	auto [FailureResult]
-		= Awaitable(&csp::systems::UserSystem::Login, UserSystem, "", DefaultLoginEmail, DefaultLoginPassword, false).Await(RequestPredicate);
-
-	EXPECT_EQ(FailureResult.GetResultCode(), csp::services::EResultCode::Failed);
-	EXPECT_EQ(FailureResult.GetFailureReason(), csp::services::EResultBaseFailureReason::None);
-
-	// False Log in
-	auto [SuccessResult]
-		= Awaitable(&csp::systems::UserSystem::Login, UserSystem, "", DefaultLoginEmail, DefaultLoginPassword, false).Await(RequestPredicate);
-
-	EXPECT_EQ(SuccessResult.GetResultCode(), csp::services::EResultCode::Failed);
-	EXPECT_EQ(SuccessResult.GetFailureReason(), csp::services::EResultBaseFailureReason::None);
+	LogIn(UserSystem,
+		  UserId,
+		  DefaultLoginEmail,
+		  DefaultLoginPassword,
+		  false,
+		  csp::services::EResultCode::Failed,
+		  csp::services::EResultBaseFailureReason::Unknown);
+
+	// true Log in
+	LogIn(UserSystem,
+		  UserId,
+		  DefaultLoginEmail,
+		  DefaultLoginPassword,
+		  true,
+		  csp::services::EResultCode::Success,
+		  csp::services::EResultBaseFailureReason::None);
 
 	LogOut(UserSystem);
```

Comment:
Please can we also test the failure codes are the expected values when we pass nullptr, true and false, indivually?